### PR TITLE
feat: comdex parse cmd for `x/wasm codes`

### PR DESCRIPTION
**Pull request by huichiaotsou**
## Description

Closes: #XXXX
jira: https://forbole.atlassian.net/browse/BDU-716

once the PR is merged and deployed, needs to:
- drop `wasm_params` & `wasm_code` tables and re-create them
- run `bdjuno parse wasm codes` to get back the contract codes
- run parse blocks cmd for the below heights to get back the contracts: 
  - `bdjuno parse blocks all --start 5212421 --end 5212421 --force`
  - `bdjuno parse blocks all --start 5212454 --end 5212454 --force`
  - `bdjuno parse blocks all --start 5212479 --end 5212479 --force`
  - `bdjuno parse blocks all --start 5212519 --end 5212519 --force`
  - `bdjuno parse blocks all --start 5272706 --end 5272706 --force`

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [x] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: cmd/parse/wasm/code.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+			height, err := db.GetLastBlockHeight()
+			if err != nil {
+				return err
+			}
+
+			wasmModule := wasm.NewModule(sources.WasmSource, parseCtx.EncodingConfig.Marshaler, db)
+
+			wasmCodes, err := wasmModule.GetWasmCodes(height)
+			if err != nil {
+				return fmt.Errorf("error while wasm codes: %s", err)
```
Comment (MonikaCat) [inline cmd/parse/wasm/code.go]:
```suggestion
				return fmt.Errorf("error while getting wasm codes: %s", err)
```

File: database/wasm.go [hunk 2]
```diff
@@ -45,6 +44,7 @@ INSERT INTO wasm_code(sender, byte_code, instantiate_permission, code_id, height
 VALUES `
```
Comment (MonikaCat) [inline database/wasm.go:48]:
```suggestion

	if len(wasmCodes) == 0 {
		return fmt.Errorf("wasm codes list is empty")
	}
	
```

**Comment by MonikaCat** [comment]
Hey @huichiaotsou can you also remove `max_wasm_code_size` inside [WasmParams type ](https://github.com/forbole/bdjuno/blob/aaron/comdex_wasm_fix/database/types/wasm.go#L45), update hasura metadata and add wasm tests pls? Thx ^^

File: database/types/wasm.go [hunk 3]
```diff
@@ -202,6 +155,6 @@ func (a WasmExecuteContractRow) Equals(b WasmExecuteContractRow) bool {
 		a.RawContractMessage == b.RawContractMessage &&
 		a.Funds.Equal(a.Funds) &&
```
Comment (MonikaCat) [inline database/types/wasm.go]:
```suggestion
		a.Funds.Equal(b.Funds) &&
```

File: database/types/wasm.go [hunk 4]
```diff
@@ -156,7 +107,7 @@ func (a WasmContractRow) Equals(b WasmContractRow) bool {
...
```
Comment (MonikaCat) [inline database/types/wasm.go]:
```suggestion
		a.Funds.Equal(b.Funds) &&
```

**Comment by MonikaCat** [comment]
@huichiaotsou hey can you also fix the lint error pls? ^^

**Review by MonikaCat — APPROVED** [review]

**Review by rmontagnin — APPROVED** [review]
utACK

**Pull request merged**